lib/Server.js

```diff
@@ -125,9 +125,12 @@ class Server {
   start() {
     const hostname = this.hostname;
     const port = this.port;
-    this.httpServer = this.server.listen(port, function() {
-      console.log('Server listening at http://%s:%s', hostname, port);
-    });
+    return new Promise(function (resolve) {
+      this.httpServer = this.server.listen(port, function() {
+        console.log('Server listening at http://%s:%s', hostname, port);
+        this._readyPromise.then(resolve);
```

Comment:
Hmm, I'm thinking of actually reworking the `_readyPromise` thing so that `new Server()` does not start the packager & webpack server right away, and instead we start everything here in `server.start()`. I think that would be more intuitive.

I'll try to get that done tonight, then I'll pull in your changes :smile: